dev-libs/xxhash/xxhash-0.6.4.recipe

```diff
@@ -0,0 +1,64 @@
+SUMMARY="A fast non-cryptographic hash algorithm"
+DESCRIPTION="xxHash is an extremely fast non-cryptographic hash algorithm, \
+working at speeds close to RAM limits."
+HOMEPAGE="https://cyan4973.github.io/xxHash/
+	http://www.xxhash.com/"
+COPYRIGHT="2012-2016 Yann Collet"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="http://github.com/Cyan4973/xxHash/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="4570ccd111df6b6386502791397906bf69b7371eb209af7d41debc2f074cdb22"
+SOURCE_FILENAME="xxHash-$portVersion.tar.gz"
+SOURCE_DIR="xxHash-$portVersion"
+PATCHES="xxHash-$portVersion.patch"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	libxxhash$secondaryArchSuffix = $portVersion
+	cmd:xxh32sum$secondaryArchSuffix = $portVersion
+	cmd:xxh64sum$secondaryArchSuffix = $portVersion
+	cmd:xxhsum$secondaryArchSuffix = $portVersion
+	lib:libxxhash$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libxxhash${secondaryArchSuffix}_devel = $portVersion
+	devel:libxxhash$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libxxhash$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage libxxhash$secondaryArchSuffix \
+	"$binDir"/xxh32sum "$binDir"/xxh64sum "$binDir"/xxhsum \
+	"$libDir/libxxhash.so.$libVersion"
+
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make BINDIR="$binDir" INCLUDEDIR="$includeDir" LIBDIR="$libDir" \
```

Comment:
```diff
-	make BINDIR="$binDir" INCLUDEDIR="$includeDir" LIBDIR="$libDir" \
+	make BINDIR="$commandBinDir" INCLUDEDIR="$includeDir" LIBDIR="$libDir" \
 		MANDIR="$manDir/man1" install
+
+	if [ "$provideRuntime" != yes ]; then
+		rm -rf "$commandBinDir" "$manDir"
+	fi
```

Since:
1) xxhash has been optimized for speed, and
2) the test suite does not build with gcc2,

it makes sense to discard the xxhsum built with gcc2 and, instead provide on x86_gcc2 the `cmd:xxhsum` built for x86 secondary arch.

Reply:
Fixed.